corehq/ex-submodules/casexml/apps/case/xform.py

```diff
@@ -272,12 +273,42 @@ def close_extension_cases(case_db, cases, device_id):
 
 
 def get_all_extensions_to_close(domain, cases):
-    if not EXTENSION_CASES_SYNC_ENABLED.enabled(domain):
+    if not toggles.EXTENSION_CASES_SYNC_ENABLED.enabled(domain):
         return set()
+    else:
+        return get_extension_cases_to_close(domain, cases)
+
+
+@extension_point(result_format=ResultFormat.SET_FLATTEN)
+def get_extension_cases_to_close(domain, cases):
     case_ids = [case.case_id for case in cases if case.closed]
     return CaseAccessors(domain).get_extension_chain(case_ids, include_closed=False)
 
 
+@get_extension_cases_to_close.extend(domains=settings.USH_CUSTOM_EXTENSION_DOMAINS)
+def get_ush_extension_cases_to_close(domain, cases):
```

Comment:
@orangejenny I am still wondering if we should enabled this based on a flag or localsetting.
Also, I couldn't find any custom USH module. Shall I create one and move this code to a `custom.ush` directory?

Reply:
I think a flag would be better if that's possible, so as not to require a code change when a new domain is added. There are a lot of USH domains, with new ones still being added somewhat regularly for testing purposes.

There's a `custom.covid` directory, this can go there.

Reply:
hmm, that could be tricky - extension points don't currently support a dynamic list of domains. @millerdev any thoughts on how to enable this workflow (or if we should)?

We could support passing in a toggle name when extending the extension point and only including it if the toggle is enabled for the domain.

Reply:
I think the feature flag use case makes this not a great candidate to be an extension point. It seems simpler to do something like
```py
from custom.covid.cases import get_ush_extension_cases_to_close

def get_extension_cases_to_close(domain, cases):
    if toggles.EXTENSION_CASES_SYNC_ENABLED.enabled(domain):
        return get_ush_extension_cases_to_close(domain, cases)

    # Made-up method name. Not sure what should be done in the default case. Return empty set?
    return get_default_extension_cases_to_close(domain, cases)
```

> thoughts on how to enable this workflow (or if we should)?

I'm not sure we should. However, if we did it might make sense to allow a domain-scoped toggle _instance_ to be passed in the extension point `domains` list. Something like
```py
@get_extension_cases_to_close.extend(domains=[toggles.EXTENSION_CASES_SYNC_ENABLED])
def get_ush_extension_cases_to_close(domain, cases):
    ...
```
`Extension.should_call_for_domain()` would check if the toggle is enabled for the domain for each toggle in the domains list.

That way domain names are easily distinguished from toggles by their type. This is kinda kludgy though because it makes the `.extend(domains=...)` semantics more complex.